testing/tests/DevExpress.ui.widgets.form/form.tests.js

```diff
@@ -97,6 +101,87 @@ QUnit.testInActiveWindow('Form\'s inputs saves value on refresh', function(asser
+        { screenWidth: 1500, expectedSize: 'lg' },
+        { screenWidth: 1000, expectedSize: 'md' },
+        { screenWidth: 900, expectedSize: 'sm' },
+        { screenWidth: 700, expectedSize: 'xs' },
+    ].forEach((testConfig) => {
+        QUnit.test(`Default implementation of screenByWidth. Screen size: ${testConfig.screenWidth}`, function(assert) {
+            const getDocumentElementStub = sinon.stub(domAdapter, 'getDocumentElement').returns({
+                clientWidth: testConfig.screenWidth
+            });
+
+            const config = {
+                colCountByScreen: { xs: 1, sm: 2, md: 3, lg: 4 },
+                items: [
+                    { dataField: 'field1' }, { dataField: 'field2' }, { dataField: 'field3' }, { dataField: 'field4' }
+                ]
+            };
+
+            const $form = $('#form').dxForm(config);
+
+            const colsCount = getColsCountFromDOM($form);
+            assert.equal(colsCount, config.colCountByScreen[testConfig.expectedSize], 'form has correct columns count');
+            getDocumentElementStub.restore();
+        });
+    });
+
+    ['defaultFunction', 'globalOption', 'instanceOption'].forEach((optionType) => {
+        ['xs', 'sm', 'md', 'lg'].forEach(screenSize => {
+            QUnit.test(`Setting screen by width. Use ${optionType}, screenSize: ${screenSize}`, function(assert) {
+                const globalOptionStub = sinon.stub().returns(screenSize);
+                const instanceOptionStub = sinon.stub().returns(screenSize);
+                const defaultFunctionStub = sinon.stub(windowModule, 'defaultScreenFactorFunc').returns(screenSize);
+
+                const config = {
+                    colCountByScreen: { xs: 1, sm: 2, md: 3, lg: 4 },
+                    items: [
+                        { dataField: 'field1' }, { dataField: 'field2' }, { dataField: 'field3' }, { dataField: 'field4' }
+                    ]
+                };
+
+                if(optionType === 'globalOption') {
+                    Form.defaultOptions({
+                        options: {
+                            screenByWidth: globalOptionStub
+                        }
+                    });
+                } else if(optionType === 'instanceOption') {
+                    config['screenByWidth'] = instanceOptionStub;
+                }
+
+
+                const $form = $('#form').dxForm(config);
+                assert.equal(globalOptionStub.called, optionType === 'globalOption', 'global function is called');
+                assert.equal(instanceOptionStub.called, optionType === 'instanceOption', 'instance function is called');
+                assert.equal(defaultFunctionStub.called, optionType === 'defaultFunction', 'default function is called');
+
+                const colsCount = getColsCountFromDOM($form);
+                assert.equal(colsCount, config.colCountByScreen[screenSize], 'form has correct columns count');
+
+                Form._classCustomRules = [];
```

Comment:
_classCustomRules ?

Reply:
It seems, it is the only possible way to restore previous behavior (after setting the Form.defaultOptions)

Reply:
It can affect on the other tests in the suite. I think that will be better to save the defaultOption config before test and reset it after test was executed.

Reply:
"Form.defaultOptions(rule)" to set options and "Form._classCustomRules = xxx;" to reset :-(
ok